Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp

```diff
@@ -316,6 +317,16 @@ void FunctionDefinitionWriter::visit(AST::UnaryExpression& unary)
     visit(unary.expression());
 }
 
+void FunctionDefinitionWriter::visit(AST::BinaryExpression& binary)
+{
+    visit(binary.lhs());
```

Comment:
No parentheses? I don't think our binary operator precedence is exactly the same as C's, IIRC.

Reply:
I think that we should eventually write these out in "assembly-as-C" form. Eg:

```t0 = <lhs>;
t1 = <rhs>;```
```t2 = t0 + t1;```

Does that make sense?

Reply:
Yes, I agree, at first I wasn't sure, but given the differences in precedence and semantics, I think that's probably easier. I think we should just implement that in a separate pass once we have more operators.